lib/sps/detail/Sps.cpp

```diff
@@ -0,0 +1,235 @@
+		}
+		*output_size += ( total_samps ) * dedispersion_plan.get_ndms()[i] * sizeof(float);
+	}
+}
+
+template<typename SpsParameterType>
+void Sps<SpsParameterType>::allocate_memory_gpu(DedispersionPlan const &dedispersion_plan)
+{
+	int time_samps = dedispersion_plan.get_t_processed()[0][0] + dedispersion_plan.get_maxshift();
+	_gpu_input_size = time_samps * (size_t) dedispersion_plan.get_nchans() * sizeof(unsigned short);
+
+	printf("\n gpu inputsize: %d\n", (int)(_gpu_input_size/1024/1024));
+	cudaError_t rc1 = ( cudaMalloc((void **)&_d_input, _gpu_input_size) );
+	if (rc1 != cudaSuccess)
+	    printf("Could not allocate memory: %d", rc1);
+
+	if (dedispersion_plan.get_nchans() < dedispersion_plan.get_max_ndms())
+	{
+		_gpu_output_size = time_samps * dedispersion_plan.get_max_ndms() * sizeof(float);
+	}
+	else
+	{
+		_gpu_output_size = time_samps * dedispersion_plan.get_nchans() * sizeof(float);
+	}
+	printf("\n gpu outputsize: %d\n", (int)(_gpu_output_size /1024/1024));
+	cudaError_t rc2 = ( cudaMalloc((void **)&_d_output, _gpu_output_size) );
+	if (rc2 != cudaSuccess)
+	    printf("Could not allocate memory: %d", rc2);
+
+
+	( cudaMemset(_d_output, 0, _gpu_output_size) );
+}
+
+template<typename SpsParameterType>
+void Sps<SpsParameterType>::operator()( unsigned device_id,
+										InputData &input_data,
+										DedispersionPlan &dedispersion_plan,
+										UserInput &user_input,
+										size_t gpu_memory,
+										std::vector<float> &output_sps,
+										float ****output_buffer,
+										size_t *output_size)
+{
+
+	//
+	long int inc = 0;
+	float tstart_local = 0.0f;
+
+	// allocate memory cpu output
+	allocate_memory_cpu_output(dedispersion_plan, output_buffer, output_size);
+	// allocate memory gpu
+	allocate_memory_gpu(dedispersion_plan);
+	//
+	output_sps.resize(*output_size/sizeof(float));
+
+	printf("\nDe-dispersing...\n");
+
+	int t, dm_range;
+	float tsamp_original = _tsamp;
+	int maxshift_original = _maxshift;
+
+	//
+	float *out_tmp = NULL;
+	out_tmp = (float *) malloc(( _t_processed[0][0] + _maxshift ) * _max_ndms * sizeof(float));
+	memset(out_tmp, 0.0f, _t_processed[0][0] + _maxshift * _max_ndms * sizeof(float));
+
+	// count the number of times that analysis is called - used to fill the output buffer of analysis
+	int analysis_call = 0;
+	for (t = 0; t < _num_tchunks; ++t)
+	{
+
+		printf("\nt_processed:\t%d, %d", _t_processed[0][t], t);
+
+		load_data(-1, _in_bin, _d_input, &_input_buffer[(long int) ( inc * _nchans )],
+							_t_processed[0][t], _maxshift, _nchans, _dmshifts);
+
+		if (user_input.get_enable_zero_dm())
+			zero_dm(_d_input, _nchans, _t_processed[0][t]+_maxshift);
+
+		corner_turn(_d_input, _d_output, _nchans, _t_processed[0][t] + _maxshift);
+		int oldBin = 1;
+
+		for (dm_range = 0; dm_range < _range; ++dm_range)
+		{
+			printf("\n\n%f\t%f\t%f\t%d", _dm_low[dm_range], _dm_high[dm_range],	_dm_step[dm_range], _ndms[dm_range]), fflush(stdout);
```

Comment:
if your going to keep them add more informatin innto the comment to say what these numbers are